### PR TITLE
Add LLL chapter in the documentation

### LINKED ISSUE #5393: void

### PR DESCRIPTION
**Pull request by axic**
Closes #5393.

### CODE REVIEW

File: docs/lll.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
+###
+LLL
+###
+
+.. _lll:
+
+LLL is a low-level language using s-expressions for the EVM.
```
Comment (ekpyron) [inline docs/lll.rst]: I'd reorder ``LLL is a low-level language for the EVM based on s-expressions.``

File: docs/lll.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+The Solidity repository contains an LLL compiler, which shares the assembler subsystem with Solidity.
+However, apart from maintaining that it still compiles, no other improvements are made to it.
+
+It is not built unless specifically requested:
```
Comment (ekpyron) [inline docs/lll.rst]: That's not yet the case, though, is it? But I guess we can do it separately afterwards.

File: docs/lll.rst [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+```bash
+$ cmake -DLLL=ON ..
+$ cmake --build
+```
+
+.. warning::
+
+    The LLL codebase is deprecated in Solidity and will be removed in the future.
```
Comment (ekpyron) [inline docs/lll.rst]: ``The LLL codebase is deprecated and will be removed from the Solidity repository in the future.``?

**Review by ekpyron — DISMISSED** [review]
Looks fine, only minor comments.

File: docs/lll.rst [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+.. _lll:
+
+LLL is a low-level language using s-expressions for the EVM.
+
+The Solidity repository contains an LLL compiler, which shares the assembler subsystem with Solidity.
+However, apart from maintaining that it still compiles, no other improvements are made to it.
+
+It is not built unless specifically requested:
+```bash
+$ cmake -DLLL=ON ..
```
Comment (ekpyron) [inline docs/lll.rst]: Looking at the CI build artifact, it seems like there are no linebreaks here (https://51302-40892817-gh.circle-artifacts.com/0/docs-html/lll.html). And it doesn't understand the ``bash`` part...

File: docs/lll.rst (hunk 2 above)
Comment (axic) [reply docs/lll.rst]: With #5410 it is, marked as a dependency.

File: docs/lll.rst (hunk 3 above)
Comment (axic) [reply docs/lll.rst]: Fixed.

File: docs/lll.rst (hunk 4 above)
Comment (axic) [reply docs/lll.rst]: Fixed.

File: docs/lll.rst [hunk 5]
```diff
@@ -0,0 +1,21 @@
+###
+LLL
+###
+
+.. _lll:
+
+LLL is a low-level language for the EVM based on s-expressions.
```
Comment (axic) [inline docs/lll.rst]: Maybe: with an s-expressions syntax.

File: docs/lll.rst (hunk 2 above)
Comment (ekpyron) [reply docs/lll.rst]: Ah sorry - didn't see that.

File: docs/lll.rst (hunk 5 above)
Comment (lazaridiscom) [reply docs/lll.rst]: Possibly add "(Lisp Like Language")

File: docs/index.rst [hunk 6]
```diff
@@ -88,3 +88,4 @@ Contents
    bugs.rst
    contributing.rst
    frequently-asked-questions.rst
+   lll.rst
```
Comment (chriseth) [inline docs/index.rst:91]: Do we really want to add another high level entry to the toc?

Comment (lazaridiscom) [reply docs/index.rst]: personally i would prefere to find this info right where the question "What is this LLL thing???" raises (within the code). The text is fine, it clarifies everything.

File: docs/lll.rst (hunk 5 above)
Comment (axic) [reply docs/lll.rst]: It has like 4-5 acronyms, one other: `Little Lovely Language`.

I'd avoid it in this PR.

File: docs/index.rst (hunk 6 above)
Comment (axic) [reply docs/index.rst:91]: > Do we really want to add another high level entry to the toc?
 
Yes, that is the easy way to find it. I hope we remove all of LLL with 0.6.0.

**Review by ekpyron — APPROVED** [review]

**Review by chriseth — APPROVED** [review]

File: docs/lll.rst (hunk 5 above)
Comment (lazaridiscom) [reply docs/lll.rst]: No problem, its fine as is, just to clarify:

* I don't even bother to assess "Little Lovely"
* "Lisp Like" clarifies things

Comment (axic) [reply docs/lll.rst]: I think the `s-expressions` part clarifies that it is "lisp like", but feel free to open a PR with changes to the LLL chapter.

Comment (lazaridiscom) [reply docs/lll.rst]: Nah, for me its ok as is! Thanks for the quick processing!

**Pull request merged**